### PR TITLE
Core/Missings scripts

### PR DESCRIPTION
**Pull request by zTerragor**
Core/Missings scripts - wow token

### CODE REVIEW

**Comment by zTerragor** [comment]
wow tokens:
`ScriptName 'wow_token_1' exists in database, but no core script found!
ScriptName 'wow_token_10' exists in database, but no core script found!
ScriptName 'wow_token_2' exists in database, but no core script found!
ScriptName 'wow_token_5' exists in database, but no core script found!`

File: src/server/scripts/Custom/wow_token.cpp [hunk 1]
```diff
@@ -0,0 +1,176 @@
+#include "ScriptMgr.h"
+#include "GridNotifiersImpl.h"
```
Comment (leelf00) [inline src/server/scripts/Custom/wow_token.cpp:2]: Seems not use database function.

File: src/server/scripts/Custom/wow_token.cpp [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+#include "DatabaseEnv.h" // borrar si no necesitas base de datos
```
Comment (leelf00) [inline src/server/scripts/Custom/wow_token.cpp:3]: Seems not using databases function.

File: src/server/scripts/Custom/wow_token.cpp [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+class wow_token_1 : public ItemScript
+{
+public:
+    wow_token_1() : ItemScript("wow_token_1") {}
+
+    bool OnUse(Player *player, Item *item, const SpellCastTargets &)
+    {
+        if (player->IsInCombat() || player->InArena() || player->InBattleground()) //Item is not usable in combat, arenas and battlegrounds. This can be modified to your taste.
+        {
+            player->GetSession()->SendNotification("You may not use this token whilst you are in combat or present in an arena or battleground.");
```
Comment (leelf00) [inline src/server/scripts/Custom/wow_token.cpp:23]: I think custom message can handler in trinity_string table.

**Comment by zTerragor** [comment]
it was originally in the code, but alexYa was removed. but not completely. we have 2 ways to remove completely or or add missing code fragments

File: src/server/scripts/Custom/wow_token.cpp [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+                ChatHandler(player->GetSession()).PSendSysMessage("You do not have the necessary token.");
+            }
+        }
+        return true;
+    }
+};
+
+void AddSC_wow_token() // Add to scriptloader normally
+{
+    new wow_token_1();
```
Comment (leelf00) [inline src/server/scripts/Custom/wow_token.cpp:172]: Maybe token_1,token_2,token_5,token_10 can handler in onescript.

**Comment by alexkulya** [comment]
Isn't this variant easier?
https://github.com/alexkulya/pandaria_5.4.8/blob/master/src/server/scripts/Custom/wow_token.cpp

**Review by fuzionproject — APPROVED** [review]

**Comment by leelf00** [comment]
This commit already merge into master branch.

**Pull request closed without merging**